finality-aleph/src/data_io/data_provider.rs

```diff
@@ -296,3 +309,126 @@ impl<B: BlockT> aleph_bft::DataProvider<AlephData<B>> for DataProvider<B> {
         data
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        data_io::{
+            data_provider::{ChainTracker, ChainTrackerConfig},
+            AlephData, MAX_DATA_BRANCH_LEN,
+        },
+        testing::{client_chain_builder::ClientChainBuilder, mocks::aleph_data_from_blocks},
+        SessionBoundaries, SessionId, SessionPeriod,
+    };
+    use futures::channel::oneshot;
+    use std::future::Future;
+    use std::{sync::Arc, time::Duration};
+    use substrate_test_runtime_client::{
+        runtime::Block, DefaultTestClientBuilderExt, TestClientBuilder, TestClientBuilderExt,
+    };
+    use tokio::time::sleep;
+
+    const SESSION_LEN: u32 = 100;
+    // The lower the interval the less time the tests take, however setting this too low might cause
+    // the tests to fail. Even though 1ms works with no issues, we set it to 5ms for safety.
+    const REFRESH_INTERVAL_MILLIS: u64 = 5;
```

Comment:
maybe just a duration instead of primitive type with unit suffix? if I'm not mistaken, `Duration` works fine with `const`